test/framework/options.py

```diff
@@ -2968,6 +2968,51 @@ def test_list_software(self):
         self.assertTrue(re.search('^\s+\* GCC v4.6.3: dummy', txt, re.M))
         self.assertFalse(re.search('gzip', txt, re.M))
 
+    def test_parse_optarch(self):
+        """Test correct parsing of optarch option."""
+        
+        # Check for EasyBuildErrors
+        args = ['--optarch=Intel:something;GCC']
+        error_msg = "The optarch option has an incorrect syntax"
+        self.assertErrorRegex(EasyBuildError, error_msg, self.eb_main, args, raise_error=True)
+
+        args = ['--optarch=Intel:something;GCC:somethingelse;Intel:anothersomething']
+        error_msg = "The optarch option contains duplicated entries for compiler"
+        self.assertErrorRegex(EasyBuildError, error_msg, self.eb_main, args, raise_error=True)
+
+        # Check that it doesn't raise an exception when the input is correct
+        test_easyconfig = 'easyconfigs/test_ecs/t/toy/toy-0.0.eb'
+        test_cases = [
+                ['--optarch=',test_easyconfig],
+                ['--optarch=something',test_easyconfig],
+                ['--optarch=GENERIC',test_easyconfig],
+                ['--optarch=Intel:something',test_easyconfig],
+                ['--optarch=Intel:something;GCC:somethingelse',test_easyconfig],
+                ['--optarch=Intel:GENERIC;GCC:somethingelse',test_easyconfig],
+                ['--optarch=Intel:;GCC:somethingelse;',test_easyconfig]
+                ]
+        for args in test_cases:
+            self.eb_main(args, raise_error=True)
```

Comment:
Doing a 'full' run using `toy-0.0.eb` is meaningless imho, since `toolchain.prepare` will never be called since the `dummy` toolchain is used.

So, yes, I would drop this, since this test is mainly about how the `optarch` value is being parsed.

I would also change the tests where you check for the error on faulty output to not use `eb_main` but `options.postprocess()`?